pkg/splunk/enterprise/indexercluster.go

```diff
@@ -213,17 +367,43 @@ func (mgr *indexerClusterPodManager) decommission(n int32, enforceCounts bool) (
 
 // getClient for indexerClusterPodManager returns a SplunkClient for the member n
 func (mgr *indexerClusterPodManager) getClient(n int32) *splclient.SplunkClient {
+	// Get Pod Name
 	memberName := GetSplunkStatefulsetPodName(SplunkIndexer, mgr.cr.GetName(), n)
+
+	// Get Fully Qualified Domain Name
 	fqdnName := splcommon.GetServiceFQDN(mgr.cr.GetNamespace(),
 		fmt.Sprintf("%s.%s", memberName, GetSplunkServiceName(SplunkIndexer, mgr.cr.GetName(), true)))
-	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", string(mgr.secrets.Data["password"]))
+
+	// Retreive admin password from Pod
+	adminPwd, err := splutil.GetSpecificSecretTokenFromPod(mgr.c, memberName, mgr.cr.GetNamespace(), "password")
+	if err != nil {
+		fmt.Errorf("Couldn't retreive the admin password from secret data %s", err.Error())
+	}
+
+	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", adminPwd)
 }
 
 // getClusterMasterClient for indexerClusterPodManager returns a SplunkClient for cluster master
 func (mgr *indexerClusterPodManager) getClusterMasterClient() *splclient.SplunkClient {
-	masterIdxcName := mgr.cr.Spec.ClusterMasterRef.Name
+	// Retreive admin password from Pod
+	var masterIdxcName string
+	if len(mgr.cr.Spec.ClusterMasterRef.Name) > 0 {
+		masterIdxcName = mgr.cr.Spec.ClusterMasterRef.Name
+	} else {
+		mgr.log.Info("Empty cluster master reference")
+	}
+
+	// Get Fully Qualified Domain Name
 	fqdnName := splcommon.GetServiceFQDN(mgr.cr.GetNamespace(), GetSplunkServiceName(SplunkClusterMaster, masterIdxcName, false))
-	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", string(mgr.secrets.Data["password"]))
+
+	// Retreive admin password for Pod
```

Comment:
super duper nit: %s/Retreive/Retrieve :)

Reply:
Update all places for typo